### PR TITLE
Make reservation from table reaction

### LINKED ISSUE #65: Implement MakeReservationFromTable reaction handler
We need to implement the `MakeReservationFromTable` reaction handler for our Splendor mobile game WebSocket server. This handler should handle player requests to make a reservation from the table, and only allow it if it is their turn. Upon receiving a valid request, the server should send an announcement to all players in the game. The implementation should update the game state in the database. If the request is invalid, the server should only send a response about the error to the requester.

More info in the java doc of this class.
https://github.com/Splendor-Mobile-Game/ServerSide/blob/7bf308007d5896d58199e58e5a6157ae6235b744/src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java#L9-L49

### PR DESCRIPTION
**Pull request by Antonidass5**
finished make reservation from table reaction 🤙

### CODE REVIEW

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java [hunk 1]
```diff
@@ -54,9 +75,164 @@ public MakeReservationFromTable(int connectionHashCode, UserMessage userMessage,
...
+            this.ruby = ruby;
+            this.emerald = emerald;
+            this.sapphire = sapphire;
+            this.diamond = diamond;
+            this.onyx = onyx;
+        }
+    }
+    public class ResponseData{
+        public UUID userUuid;
+        public CardDataResponse card;
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java]: Change the name of this card field, it's not clear what it is. Also please update documentation of this class that is before class declaration.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java [hunk 2]
```diff
@@ -54,9 +75,164 @@ public MakeReservationFromTable(int connectionHashCode, UserMessage userMessage,
...
+        public boolean goldenToken;
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java]: Maybe `gotGoldenToken` would be more meaningful name

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java [hunk 3]
```diff
@@ -54,9 +75,164 @@ public MakeReservationFromTable(int connectionHashCode, UserMessage userMessage,
...
+            throw new InvalidUsernameException("It's not this user turn");
+        //TODO make exception
+
+        //Check if limit of reserved card is reached for player
+        if(database.getUser(dataDTO.userDTO.uuid).getReservationCount()>=3)
+            throw new UserReservationException("User has to many reserved cards in hand");
+
+        //Check if limit of reserved card is reached for game
+        if(database.getUser(dataDTO.userDTO.uuid).getThroughoutGameReservationCount()>=5)
+            throw new UserReservationException("User reached the limit of reserved cards per game");
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromTable.java:233]: This isn't proper check. Rules of the game say something else. This is a bug and this method was introduced earlier by someone else. I will write a new issue about fixing it. So currently leave it as it is.

Rules say, there can be only 5 cards reserved in a single moment. Not that one player can reserve only 5 cards throughout the entire game.

**Review by kestivvi — CHANGES_REQUESTED** [review]

**Review by kestivvi — DISMISSED** [review]
Superb 💎

**Review by Vertonowsky — DISMISSED** [review]

**Review by kestivvi — APPROVED** [review]

**Review by walczak1404 — APPROVED** [review]

**Pull request merged**